docs/whats-new/preview-4.md

```diff
@@ -0,0 +1,703 @@
+
+### Metrics table view
+
+With preview 4, we have introduced a screen reader compatible table view for display of metrics data. This has similar options, as the graph, for filters and selecting the duration of time range for metrics display. The default data display is set to "Only show value updates" and can be toggled to display all data points.
+
+![image](https://github.com/dotnet/docs-aspire/assets/102772054/4622cf3a-dcf2-4149-9636-d1bef8184c5c)
+
+## Databases, EntityFramework and Aspire
+
+### New Enrich methods
+
+Preview 4 introduces new methods for configuring Entity Framework. The existing `Add[Provider]DbContext()` methods used to register and configure `DbContext` classes are not sufficient for advanced cases like using a different lifetime scope, using custom service types, or configuring the underlying data sources.
+
+To solve these advanced scenarios, new methods named `Enrich[Provider]DbContext()` have been added. These methods do not register the `DbContext` and expect you to do so before invoking them.
+
+Usage example:
+
+```csharp
+var connectionString = builder.Configuration.GetConnectionString("catalogdb");
+builder.Services.AddDbContextPool<CatalogDbContext>(dbContextOptionsBuilder => dbContextOptionsBuilder.UseSqlServer(connectionString));
+builder.EnrichSqlServerDbContext<CatalogDbContext>();
+```
+
+These methods will still configure command retries, health checks, logging and telemetry.
+
+### Changes to previously existing methods
+
+Since these new methods provide a simpler way to configure the `DbContext`, the already existing ones (`Add[Provider]DbContext()`) have been simplified.
+
+1- They don't provide a way to disable connection pooling through settings anymore. Instead register the `DbContext` with connection pooling disabled and call the corresponding `Enrich` method.
+2- The `int MaxRetryCount` settings were removed and replaced by a `bool Retry` flag that and uses the default settings. It is enabled by default. To change the command retries count to a custom value please configure the `DbContext` registration using the specific provider options and invoke the corresponding `Enrich` method.  
+
+### Entity Framework Migrations
+
+We've improved using [EF Core tooling to create migrations](https://learn.microsoft.com/ef/core/managing-schemas/migrations/) with Aspire apps. Before, EF Core tooling would fail with an error that the database connection string is missing. Aspire validates that a connection string is provided by the host when an app starts up. However, EF Core tooling launches the app directly so the connection string is missing.
+
+In preview 4, Aspire now detects whether a project is launched with EF Core tooling and disables connection string validation.
+
+Here's a [sample application](https://learn.microsoft.com/en-us/samples/dotnet/aspire-samples/aspire-efcore-migrations/) that shows to create and apply migrations in an Aspire solution.
```

Comment:
This will be the link once https://github.com/dotnet/aspire-samples/pull/142 is merged and published.

Reply:
Thanks! We should say something about the fact that it's still not possible to run the migration in app either (maybe with the database migration page) @DamianEdwards?

Also there are issues with using the migration tooling while the app is running because files are locked and the tooling builds by default (though you can pass no build). 

Maybe we can say that we're working on a migration guidance document (point to a docs issue), where we explain all of these nuances.

Reply:
Updated